pywinauto/unittests/test_application_linux.py

```diff
@@ -56,72 +58,75 @@ def test_start_problem(self):
             self.subprocess_app = subprocess.Popen(_test_app().split(), stdout=subprocess.PIPE, shell=False)
             time.sleep(1)
-            app = Application()
-            app.connect(path=_test_app())
-            self.assertEqual(app.process, self.subprocess_app.pid)
-            app.kill()
+            self.app.connect(path=_test_app())
+            self.assertEqual(self.app.process, self.subprocess_app.pid)
 
         def test_get_cpu_usage(self):
-            app = Application()
-            app.start(_test_app())
+            self.app.start(_test_app())
             time.sleep(1)
-            self.assertGreater(app.cpu_usage(), 0)
-            app.kill()
+            self.assertGreater(self.app.cpu_usage(), 0)
 
         def test_is_process_running(self):
-            app = Application()
-            app.start(_test_app())
+            self.app.start(_test_app())
             time.sleep(1)
-            self.assertTrue(app.is_process_running())
-            app.kill()
-
-        def test_killed_app_not_running(self):
-            app = Application()
-            app.start(_test_app())
-            time.sleep(1)
-            app.kill()
-            self.assertFalse(app.is_process_running())
+            self.assertTrue(self.app.is_process_running())
+            self.app.kill()
+            self.assertFalse(self.app.is_process_running())
 
         def test_kill_killed_app(self):
-            app = Application()
-            app.start(_test_app())
+            self.app.start(_test_app())
             time.sleep(1)
-            app.kill()
-            self.assertTrue(app.kill())
+            self.app.kill()
+            self.assertTrue(self.app.kill())
 
         def test_kill_connected_app(self):
             self.subprocess_app = subprocess.Popen(_test_app().split(), stdout=subprocess.PIPE, shell=False)
             time.sleep(1)
-            app = Application()
-            app.connect(pid=self.subprocess_app.pid)
-            app.kill()
+            self.app.connect(pid=self.subprocess_app.pid)
+            self.app.kill()
 
             # Unlock the subprocess explicity, otherwise
             # it's presented in /proc as a zombie waiting for
             # the parent process to pickup the return code
             self.subprocess_app.communicate()
             self.subprocess_app = None
 
-            self.assertFalse(app.is_process_running())
+            self.assertFalse(self.app.is_process_running())
+
+
+    class WindowSpecificationTestCases(unittest.TestCase):
+
+        """Unit tests for the application.WindowSpecification class"""
+
+        def setUp(self):
+            """Set some data and ensure the application is in the state we want"""
+            self.app = Application()
+
+        def tearDown(self):
+            """Close the application after tests"""
+            self.app.kill()
+
+        def test_app_binding(self):
+            self.app.start(_test_app())
```

Comment:
Previous `self.app` instance seems to keep running after the test is finished. It's worth preventive killing it.

Reply:
Not sure I understand. Do you mean an app instance from the previous test suite: "ApplicationTestCases" ? We have `self.app.kill()` calls in tearDown methods of all test suites in this file.

Reply:
You call `self.app.start` twice: in `setUp` method and in the test. But `kill` is called once only. Did you run the tests manually on local machine? Don't you have some non-killed windows?

Reply:
Oops, sorry. `setUp` doesn't call `self.app.start`. My fault.